File: platform/src/main/java/ru/abondin/hreasy/platform/config/LdapConfigurationProperties.java
package ru.abondin.hreasy.platform.config;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

/**
 * Properties to configure LDAP
 */
@Configuration
@ConfigurationProperties("hreasy.ldap")
@Data
public class LdapConfigurationProperties {
    /**
     * Server URL
     */
    private String serverUrl = "ldap://dc.stm.local";
    /**
     * User DN to authorize
     */
    private String userDn = "CN=<admin_user>,CN=Users,DC=stm,DC=local";
    private String userPassword = "<PASSWORD>";
    private String searchBase = "OU=Internal,OU=Enterprise,DC=stm,DC=local";
    private String searchFilter = "sAMAccountName={0}";
}

File: platform/src/main/kotlin/ru/abondin/hreasy/platform/config/LdapConfigurationProperties.kt
package ru.abondin.hreasy.platform.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Configuration

/**
 * Properties to configure LDAP
 */
@Configuration
@ConfigurationProperties("hreasy.ldap")
class LdapConfigurationProperties {
    /**
     * Server URL
     */
    var serverUrl = "ldap://dc.stm.local";
    /**
     * User DN to authorize
     */
    var userDn = "CN=<admin_user>,CN=Users,DC=stm,DC=local";
    var userPassword = "<PASSWORD>";
    var searchBase = "OU=Internal,OU=Enterprise,DC=stm,DC=local";
    var searchFilter = "sAMAccountName={0}";
}